src/v2/trading.ts

```diff
@@ -0,0 +1,82 @@
+import { coroutine as co } from 'bluebird';
+import Bluebird = require('bluebird');
+
+export class Trading {
+  private bitgo;
+  private basecoin;
+  private wallet;
+
+  constructor(bitgo, basecoin, wallet) {
+    this.bitgo = bitgo;
+    this.basecoin = basecoin;
+    this.wallet = wallet;
+  }
+
+  /**
+   * Builds a payload authorizing a trade from this trading account. The currency and amount must be specified, as well as a list
+   * of trade counterparties.
+   * @param params
+   * @param params.currency the currency this wallet will be sending as part of the trade
+   * @param params.amount the amount of currency (in base units, such as cents, satoshis, or wei) authorized to be spent as part of the trade
+   * @param params.otherParties array of trading account IDs authorized to receive funds as part of the trade
+   * @param callback
+   * @returns unsigned trade payload for the given parameters
+   */
+  public buildTradePayload(params: BuildPayloadParameters, callback?): Bluebird<string> {
+    return co(function *buildTradePayload() {
+      const url = this.bitgo.microservicesUrl('/api/trade/v1/payload');
+
+      const body = {
+        accountId: this.wallet.id(),
+        currency: params.currency,
+        amount: params.amount,
+        otherParties: params.otherParties
+      };
+
+      const response = yield this.bitgo.post(url).send(body).result();
+
+      return response.payload;
+    }).call(this).asCallback(callback);
+  }
+
+  /**
+   * Builds and signs a payload authorizing a trade from this trading account. The currency and amount must be specified, as well
+   * as a list of trade counterparties. Requires the wallet keychain or raw private key to sign the transaction. Both
+   * the payload and signature are returned.
+   * @param params
+   * @param params.currency the currency this wallet will be sending as part of the trade
+   * @param params.amount the amount of currency (in base units, such as cents, satoshis, or wei) authorized to be spent as part of the trade
+   * @param params.otherParties array of trading account IDs authorized to receive funds as part of the trade
+   * @param params.walletPassphrase the wallet password, for decrypting the private key for signing
+   * @param callback
+   * @returns {{payload: string, signature: string}}
+   */
+  public signTradePayload(params: SignPayloadParameters, callback?): Bluebird<SignedPayload> {
+    return co(function *signTradePayload() {
+      const payload = yield this.buildTradePayload(params);
+
+      const key = yield this.basecoin.keychains().get({ id: this.wallet.keyIds()[0] });
+      const prv = this.bitgo.decrypt({ input: key.encryptedPrv, password: params.walletPassphrase });
+
+      const signature = this.basecoin.signMessage({ prv }, payload).toString('hex');
+
+      return { payload, signature };
+    }).call(this).asCallback(callback);
+  }
+}
+
+
+interface BuildPayloadParameters {
```

Comment:
I think these interfaces are part of the public API and thus should be exported.

Do you anticipate users of bitgojs would need to know about these types in order to use the feature?

Reply:
Decent IDEs will still offer type checking even if these interfaces can't be directly imported, but this use case wouldn't work:

```ts
import { BuildPayloadParameters } from 'bitgo';
const params: BuildPayloadParameters {
  ...
}
```

i.e. manually creating an object with that type. We can export that interface if you want, but then we'll be exporting a ton of interfaces which may get confusing. Thoughts?

Reply:
It won't work because we don't re-export any interfaces out of the top level, so you're correct in saying that they can't be directly imported in the current project setup. However, I still think we should export if only so that future developers are aware that these types are part of the public API.

Reply:
Ah okay, sounds good